runtime/mux.go

```diff
@@ -55,6 +58,19 @@ func WithForwardResponseOption(forwardResponseOption func(context.Context, http.
 	}
 }
 
+// A QueryParameterParser populates message form HTTP query parameters values
+type QueryParameterParser func(msg proto.Message, values url.Values, filter *utilities.DoubleArray) error
```

Comment:
What do you think about making this an interface?

Reply:
can do, will make the changes for global parser and see if interface fits/is needed.

Reply:
added interface in query.go